cmd/logger.go

```diff
@@ -137,7 +93,58 @@ func getSource() string {
 	return fmt.Sprintf("[%s:%d:%s()]", filename, lineNum, funcName)
 }
 
-func logIf(level logrus.Level, source string, err error, msg string, data ...interface{}) {
+// TracePoint is to capture stack trace
+type TracePoint struct {
+	Line     int    `json:"line,omitempty"`
+	Filename string `json:"file,omitempty"`
+	Function string `json:"func,omitempty"`
+}
+
+// String method - returns the string
+// representation of a TracePoint instance
+func (tp TracePoint) String() string {
+	return tp.Filename + ":" + strconv.Itoa(tp.Line) + ":" + tp.Function
+}
+
+// trace method - to create stack trace
+func trace(level int) []string {
+	var stack []string
+	var pc uintptr
+	var file string
+	var line int
+	ok := true
+	var callTrace []TracePoint
+	pc, file, line, ok = runtime.Caller(level)
+	rootPath := filepath.Dir(file)
+	for ok {
+		// Clean up the common root path, rootPath,
+		// from the collected stack trace
+		function := runtime.FuncForPC(pc).Name()
+		_, function = filepath.Split(function)
+		file = strings.TrimPrefix(file, rootPath+string(os.PathSeparator))
+		tp := TracePoint{}
+		tp = TracePoint{Line: line, Filename: file, Function: function}
+		callTrace = append(callTrace, tp)
+		level++
+		pc, file, line, ok = runtime.Caller(level)
+	}
+	for _, elem := range callTrace[:len(callTrace)-2] {
+		stack = append(stack, elem.String())
+	}
```

Comment:
I think `TracePoint` is not required as a separate type. You could directly consume line number, filename and function name in this function.

Reply:
Done.